subsys/net/lib/nrf_cloud/src/nrf_cloud_pgps.c

```diff
@@ -1201,8 +1236,17 @@ static int open_storage(uint32_t offset, bool preserve)
 		err = stream_flash_buffered_write(&stream, buf, block_offset, false);
 		if (err) {
 			LOG_ERR("Error writing back %u original bytes", block_offset);
+			goto fail;
 		}
 	}
+
+	return 0;
+
+fail:
+#if defined(CONFIG_PM_PARTITION_REGION_PGPS_EXTERNAL)
+	k_sem_give(&flash_accessible);
+	LOG_DBG("flash_accessible UNLOCKED");
+#endif
```

Comment:
Maybe there should be a dedicated lock/unlock function, with the `CONFIG_PM_PARTITION_REGION_PGPS_EXTERNAL` guard inside it, and then call it unguarded in all the appropriate places

Reply:
yes, I was rushing this to get it done in time...